component/xload/xload.go

```diff
@@ -0,0 +1,300 @@
+	return compName
+}
+
+func (xl *Xload) SetName(name string) {
+	xl.BaseComponent.SetName(name)
+}
+
+func (xl *Xload) SetNextComponent(nc internal.Component) {
+	xl.BaseComponent.SetNextComponent(nc)
+}
+
+func (xl *Xload) Priority() internal.ComponentPriority {
+	return internal.EComponentPriority.LevelMid()
+}
+
+// Configure : Pipeline will call this method after constructor so that you can read config and initialize yourself
+func (xl *Xload) Configure(_ bool) error {
+	log.Trace("Xload::Configure : %s", xl.Name())
+
+	// xload component should be used only in readonly mode
+	var readonly bool
+	err := config.UnmarshalKey("read-only", &readonly)
+	if err != nil {
+		log.Err("Xload::Configure : config error [unable to obtain read-only]")
+		return fmt.Errorf("config error in %s [%s]", xl.Name(), err.Error())
+	}
+
+	if !readonly {
+		log.Err("Xload::Configure : Xload component should be used only in read-only mode")
+		return fmt.Errorf("Xload component should be used in only in read-only mode")
+	}
+
+	conf := XloadOptions{}
+	err = config.UnmarshalKey(xl.Name(), &conf)
+	if err != nil {
+		log.Err("Xload::Configure : config error [invalid config attributes]")
+		return fmt.Errorf("Xload: config error [invalid config attributes]")
+	}
+
+	xl.blockSize = uint64(defaultBlockSize) * _1MB // 16 MB as deafult block size
+	if config.IsSet(compName + ".block-size-mb") {
+		xl.blockSize = uint64(conf.BlockSize * float64(_1MB))
+	}
+
+	xl.path = common.ExpandPath(strings.TrimSpace(conf.Path))
+	if xl.path == "" {
+		// TODO:: xload : should we use current working directory in this case
+		log.Err("Xload::Configure : config error [path not given in xload]")
+		return fmt.Errorf("config error in %s [path not given]", xl.Name())
+	} else {
+		//check mnt path is not same as xload path
+		mntPath := ""
+		err = config.UnmarshalKey("mount-path", &mntPath)
+		if err != nil {
+			log.Err("Xload::Configure : config error [unable to obtain Mount Path [%s]]", err.Error())
+			return fmt.Errorf("config error in %s [%s]", xl.Name(), err.Error())
+		}
+
+		if xl.path == mntPath {
+			log.Err("Xload::Configure : config error [xload path is same as mount path]")
+			return fmt.Errorf("config error in %s error [xload path is same as mount path]", xl.Name())
+		}
+
+		_, err = os.Stat(xl.path)
+		if os.IsNotExist(err) {
+			log.Info("Xload::Configure : config error [xload path does not exist, attempting to create path]")
+			err := os.Mkdir(xl.path, os.FileMode(0755))
+			if err != nil {
+				log.Err("Xload::Configure : config error creating directory of xload path [%s]", err.Error())
+				return fmt.Errorf("config error in %s [%s]", xl.Name(), err.Error())
+			}
+		}
+
+		if !common.IsDirectoryEmpty(xl.path) {
+			log.Err("Xload::Configure : config error %s directory is not empty", xl.path)
+			return fmt.Errorf("config error in %s [temp directory not empty]", xl.Name())
+		}
+	}
+
+	var mode xcommon.Mode
+	err = mode.Parse(conf.Mode)
```

Comment:
Mode is a mandatory parameter as of now. As we are only having download feature for now the default value shall be preload related mode.

Reply:
Setting default mode to "preload"